code/modules/mob/living/carbon/human/body_accessories.dm

```diff
@@ -0,0 +1,133 @@
+var/global/list/body_accessory_by_name = list("None" = null)
+
+/hook/startup/proc/initalize_body_accessories()
+
+	__init_body_accessory(/datum/body_accessory/body)
+	__init_body_accessory(/datum/body_accessory/tail)
+
+	if(body_accessory_by_name.len)
+		if(initialize_body_accessory_by_species())
+			return 1
+
+	return 0 //fail if no bodies are found
+
+var/global/list/body_accessory_by_species = list("None" = null)
+
+/proc/initialize_body_accessory_by_species()
+	for(var/B in body_accessory_by_name)
+		var/datum/body_accessory/accessory = body_accessory_by_name[B]
+		if(!istype(accessory))	continue
+
+		for(var/species in accessory.allowed_species)
+			if(!body_accessory_by_species["[species]"])	body_accessory_by_species["[species]"] = list()
+			body_accessory_by_species["[species]"] += accessory
+
+	if(body_accessory_by_species.len)
+		return 1
+	return 0
+
+/proc/__init_body_accessory(var/ba_path)
+	if(ispath(ba_path))
+		var/_added_counter = 0
+
+		for(var/A in subtypesof(ba_path))
+			var/datum/body_accessory/B = new A
+			if(istype(B))
+				body_accessory_by_name[B.name] += B
+				++_added_counter
+
+		if(_added_counter)
+			return 1
+	return 0
+
+
+/datum/body_accessory
+	var/name = "default"
+
+	var/icon = null
+	var/icon_state = ""
+
+	var/animated_icon = null
+	var/animated_icon_state = ""
+
+	var/blend_mode = null
+
+	var/pixel_x_offset = 0
+	var/pixel_y_offset = 0
+
+	var/list/allowed_species = list()
+
+/datum/body_accessory/proc/try_restrictions(var/mob/living/carbon/human/H)
+	return 1
+
+/datum/body_accessory/proc/get_pixel_x(var/mob/living/carbon/human/H)
```

Comment:
This... doesn't really need to exist.
same for it's Y counterpart, and actually adds a small about of lag to obtaining that var (because of the proccall)